MekHQ/src/mekhq/campaign/personnel/RetirementDefectionTracker.java

```diff
@@ -99,15 +98,15 @@ public static Money getShareValue(Campaign campaign) {
         String financialReport = campaign.getFinancialReport();
         Money netWorth = Money.zero();
         try {
-            Pattern p = Pattern.compile("Net Worth\\D*(.*)");
+            Pattern p = Pattern.compile("Net Worth\\D*([\\d,.]+)");
```

Comment:
An unescaped . matches anything, so [\\d,.]+` is the same as `.+`.

Reply:
You can thank night shift programming for this.